BJS2-96035: -add config for open api -add basic doc for work schedule controller

Deathroels opened the pull request.

Deathroels committed:
BJS2-96035:
-add config for open api
-add basic doc for work schedule controller

20cognosce commented on src/main/java/school/faang/user_service/config/OpenApiConfig.java:
```diff
@@ -0,0 +1,20 @@
+                        .title("User Service API")
+                        .description("API documentation for the User Service")
+                        .version("1.0.0"));
+    }
```
Во-первых это не обязательно, webmvc-ui по умолчанию работает, во-вторых, всего 3 параметра, ради них целый бин поднимать? было бы что-то более значимое в настройках

20cognosce commented on src/main/java/school/faang/user_service/controller/workschedule/WorkScheduleController.java:
```diff
@@ -15,28 +17,33 @@
 import school.faang.user_service.dto.workschedule.WorkScheduleDto;
 import school.faang.user_service.dto.workschedule.WorkScheduleUpdateDto;
 
+@Tag(name = "Work Schedule Controller", description = "APIs for managing work schedules")
```
Посмотри на ПР Ильи, я ему принял
Это очень скудненько конечно, ты не пощупал почти никакой функционал сваггера

и пожалуйста выносите доку отдельно - в интерфейс

20cognosce reviewed (changes requested):
Обнови версии на актуаульные в градле

Deathroels committed:
BJS2-96074:
-refactor remove class for name and version of api, moved to application.yaml

Deathroels committed:
BJS2-96074:
-moved api info in WorkScheduleApi interface

Deathroels committed:
BJS2-96074:
-update org.springframework.boot to 3.1.12

Deathroels committed:
BJS2-96074:
-it was possible to upgrade springdoc-openapi-starter-webmvc-ui to 2.6.0 as much as possible

Deathroels committed:
BJS2-96074:
-removed unused imports

20cognosce commented on src/main/java/school/faang/user_service/controller/workschedule/WorkScheduleController.java:
```diff
@@ -16,27 +16,31 @@
+public class WorkScheduleController implements WorkScheduleApi {
     private final WorkScheduleFacade workScheduleFacade;
 
+    @Override
```
не обязательно писать override вроде

20cognosce commented on src/main/java/school/faang/user_service/controller/workschedule/WorkScheduleApi.java:
```diff
@@ -0,0 +1,55 @@
+            @ApiResponse(responseCode = "400", description = "Invalid request parameters"),
+            @ApiResponse(responseCode = "404", description = "Work schedule not found")
+    })
+    WorkScheduleDto updateWorkSchedule(long workScheduleId, WorkScheduleUpdateDto workScheduleUpdateDto);
```
к параметрам можно тоже писать доку
